### PR TITLE
Clarify GC.Collect guidance in memory management doc

### PR DESCRIPTION
**Pull request by adityamandaleeka**
Improving the section about `GC.Collect` usage in "Memory management and garbage collection (GC) in ASP.NET Core".

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [aspnetcore/performance/memory.md](https://github.com/dotnet/AspNetCore.Docs/blob/4b6ef1d391f7b567d007ad3a14ed1e0393fb8884/aspnetcore/performance/memory.md) | [Memory management and garbage collection (GC) in ASP.NET Core](https://review.learn.microsoft.com/en-us/aspnet/core/performance/memory?branch=pr-en-us-30202) |

### CODE REVIEW

**Comment by adityamandaleeka** [comment]
Also fixed a typo elsewhere on the page...

File: aspnetcore/performance/memory.md [hunk 1]
```diff
@@ -30,13 +30,11 @@ When an ASP.NET Core app starts, the GC:
...
-### Call GC.Collect
+### Note about calling GC.Collect
 
-Calling [GC.Collect](xref:System.GC.Collect%2A) explicitly:
+In general, ASP.NET Core apps in production should **not** use [GC.Collect](xref:System.GC.Collect%2A) explicitly. Inducing garbage collections at sub-optimal times can decrease performance significantly.
 
-* Should **not** be done by production ASP.NET Core apps.
-* Is useful when investigating memory leaks.
-* When investigating, verifies the GC has removed all dangling objects from memory so memory can be measured.
+GC.Collect can be useful when investigating memory leaks. Calling `GC.Collect()` will trigger a blocking garbage collection cycle that will try to reclaim all objects inaccessible from managed code. It is therefore a useful way to understand the size of the reachable/live objects in the heap, and track growth of that size over time.
```
Comment (Rick-Anderson) [inline aspnetcore/performance/memory.md:37]:
```suggestion
GC.Collect is useful when investigating memory leaks. Calling `GC.Collect()` triggers a blocking garbage collection cycle that tries to reclaim all objects inaccessible from managed code. It's a useful way to understand the size of the reachable live objects in the heap, and track growth of memory size over time.
```

File: aspnetcore/performance/memory.md [hunk 2]
```diff
@@ -30,13 +30,11 @@ When an ASP.NET Core app starts, the GC:
 
 The preceding memory allocations are done for performance reasons. The performance benefit comes from heap segments in contiguous memory.
 
-### Call GC.Collect
+### Note about calling GC.Collect
```
Comment (Rick-Anderson) [inline aspnetcore/performance/memory.md:33]:
```suggestion
### GC.Collect caveats 
```

**Review by Rick-Anderson — APPROVED** [review]
remove FT (future tense)

**Comment by adityamandaleeka** [comment]
Thanks @Rick-Anderson. Feedback applied.

**Pull request merged**